src/Agent.Plugins/PipelineArtifact/PipelineArtifactPluginV2.cs

```diff
@@ -286,7 +286,14 @@ private async Task<int> GetPipelineIdAsync(AgentTaskPluginExecutionContext conte
             var isDefinitionNum = Int32.TryParse(pipelineDefinition, out int definition);
             if(!isDefinitionNum) 
             {
-                definition = (await buildHttpClient.GetDefinitionsAsync(new System.Guid(project), pipelineDefinition, cancellationToken: cancellationToken)).FirstOrDefault().Id;
+                try
+                {
+                    definition = (await buildHttpClient.GetDefinitionsAsync(new System.Guid(project), pipelineDefinition, cancellationToken: cancellationToken)).FirstOrDefault().Id;
```

Comment:
.FirstOrDefault() can cause a nullref too. You may want to test that first and then if its null throw something nicer.

Reply:
or just use .First() if that's what you want

Reply:
Acutally, yes. We just need first() here. Not sure why FirstOrDefault() was used in the original code.

Reply:
Do you really want to interpret *all* exceptions (timeouts, server side 500s, auth problems, etc) as "pipeline does not exist" though? If you are expecting to get a definition and want to raise when it isn't found specifically, using FirstOrDefault() and checking for null rather than just accessing .Id as it did originally would be better.